cypress/integration/unit/logic/schedule-preprocess/schedule-preprocess.spec.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  calculateMissingFullWeekDays,
+  daysInMonth,
+} from "../../../../../src/state/reducers/month-state/schedule-data/common-reducers";
+import { ScheduleKey } from "../../../../../src/api/persistance-store.model";
+import { numberOfWeeksInMonth } from "../../../../support/commands";
+
+interface TestCase {
+  title: string;
+  monthNumber: number;
+  year: number;
+  dayInMonth: number;
+  missingDaysToFullWeek: {
+    prevMonth: number;
+    nextMonth: number;
+  };
+}
+
+const testCases: TestCase[] = [
+  {
+    title: "january (31 days)",
+    monthNumber: 0,
+    year: 2021,
+    dayInMonth: 31,
+    missingDaysToFullWeek: {
+      prevMonth: 4,
+      nextMonth: 0,
+    },
+  },
+  {
+    title: "month consisting only of full weeks",
+    monthNumber: 1,
+    year: 2021,
+    dayInMonth: 28,
+    missingDaysToFullWeek: {
+      prevMonth: 0,
+      nextMonth: 0,
+    },
+  },
+  {
+    title: "february with leap day (29 days)",
+    monthNumber: 1,
+    year: 2020,
+    dayInMonth: 29,
+    missingDaysToFullWeek: {
+      prevMonth: 5,
+      nextMonth: 1,
+    },
+  },
+];
+
+describe("Schedule preprocessor functions", () => {
+  testCases.forEach((test) => {
+    it(`Should calculate proper month length for ${test.title}`, () => {
+      const days = daysInMonth(test.monthNumber, test.year);
+      cy.log(`${days}`);
```

Comment:
nie wiem czy log jest potrzebny, mamy już nazwę testu, która wszystko wyjaśnia, jak będzie bug, to wiadomo w którym teście

Reply:
nie potrzebny, zapomniałem o usunąć